### PR TITLE
Support for Well Specification (WELSPECS) reporting

### PR DESCRIPTION
**Pull request by wito**
This feature adds support for producing Well Specification reports during simulation runs.

### CODE REVIEW

**Comment by wito** [comment]
jenkins build this please

File: CMakeLists_files.cmake [hunk 1]
```diff
@@ -275,6 +275,8 @@ if(ENABLE_ECL_OUTPUT)
           src/opm/output/eclipse/RestartValue.cpp
           src/opm/output/eclipse/WriteInit.cpp
           src/opm/output/eclipse/WriteRFT.cpp
+          src/opm/output/eclipse/WriteRPT.cpp
+          src/opm/output/eclipse/WriteRPT_WELSPECS.cpp
```
Comment (joakim-hove) [inline CMakeLists_files.cmake:279]: I suggest you create a directory `..eclipse/report` and then a file `Welspecs.cpp` there

Comment (wito) [reply CMakeLists_files.cmake:279]: 👍

**Comment by joakim-hove** [comment]
It turns out that we also must add some logic so that the reports are only created if there have been updates in the underlying data - that will need to be some new logic in the Schedule class.

**Comment by wito** [comment]
> It turns out that we also must add some logic so that the reports are only created if there have been updates in the underlying data - that will need to be some new logic in the Schedule class.

How about a method in `DynamicState<T>`, something like `bool is_new_state(std::size_t time_step)` which returns `true` if the value at `time_step` is introduced at that time step?

**Comment by joakim-hove** [comment]
> How about a method in DynamicState<T>, something like bool is_new_state(std::size_t time_step) which returns true if the value at time_step is introduced at that time step?

No - that was my idea!

**Comment by wito** [comment]
jenkins build this please

**Comment by wito** [comment]
jenkins build this please

**Comment by bska** [comment]
This needs a follow-up change.  Master is mostly useless now because WELSPECS is printed to the main command window instead of the .PRT file.  We should probably use the logging system here.  I'm also not too impressed by the fact that we output this information&mdash;which is mostly unchanging&mdash;on every timestep rather than limited to report-steps only.

**Comment by joakim-hove** [comment]
Hmmm - the intention was that it should not output anything at all just yet - if that is wrong a fixuå should come swiftly.

Yes it should go to the `PRT` file, and yes the update frequency needs to be updated - that being said the exact semantics for what to output is not entirely clear to me yet.

**Comment by bska** [comment]
> Hmmm - the intention was that it should not output anything at all just yet

I see.  Well, at least [this loop](https://github.com/OPM/opm-common/blob/56745f57ac9a1a896baa64a16096f27b6ff95f79/src/opm/output/eclipse/EclipseIO.cpp#L262-L264) ends up outputting copious reports to `std::cout` in one of my local models&mdash;a model that happens to have
```
RPTSCHED
  'RESTART=1' 'FIP=3' 'WELSPECS' 'WELLS=0' 'NEWTON=1' 'SUMMARY=1' 'CPU=1' /
```
For now I've just disabled that output loop locally.

**Pull request merged**